src/AutoRest.CSharp/Common/Generation/Types/TypeFactory.cs

```diff
@@ -28,7 +28,9 @@ public TypeFactory(OutputLibrary library)
             _library = library;
         }
 
-        public CSharpType CreateType(Schema schema, bool isNullable) => schema switch
+        public CSharpType CreateType(Schema schema, bool isNullable) => CreateType(schema, schema.Extensions, isNullable);
+
+        public CSharpType CreateType(Schema schema, RecordOfStringAndAny? schemaExtensions, bool isNullable) => schema switch
```

Comment:
The parameter name `schemaExtensions` here is quite misleading which indicates this extension is from the schema, and the schema has a property `Extensions`.
But in fact this method is for the case that we are using an extension from other things (like the `requestParameter` instead of the schema
How about we rename this parameter to `extensions`, and add some xml document on this method?